packages/flutter/lib/src/painting/beveled_rectangle_border.dart

```diff
@@ -0,0 +1,143 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'basic_types.dart';
+import 'border_radius.dart';
+import 'borders.dart';
+import 'edge_insets.dart';
+
+/// A rectangular border with flattened or "beveled" corners.
+///
+/// The line segments that connect the rectangle's four sides will
+/// begin and at locations offset by the corresponding border radius,
+/// but not farther than the side's center. If all the border radii
+/// exceed the sides' half widths/heights the resulting shape is
+/// diamond made by connecting the centers of the sides.
+class BeveledRectangleBorder extends ShapeBorder {
+  /// Creates a border like a [RoundedRectangleBorder] except that the corners
+  /// are joined by straight lines instead of arcs.
+  ///
+  /// The arguments must not be null.
+  const BeveledRectangleBorder({
+    this.side: BorderSide.none,
+    this.borderRadius: BorderRadius.zero,
+  }) : assert(side != null),
+       assert(borderRadius != null);
+
+  /// The style of this border.
+  final BorderSide side;
+
+  /// The radii for each corner.
```

Comment:
you should probably explain what this actually means for a beveled border, because it's not intuitive.